xcdat/spatial.py

```diff
@@ -476,6 +509,32 @@ def _get_latitude_weights(
         weights = self._calculate_weights(d_bounds)
         return weights
 
+    def _get_vertical_weights(
+        self, domain_bounds: xr.DataArray, region_bounds: Optional[np.ndarray]
+    ) -> xr.DataArray:
+        """Gets weights for the vertical axis.
+
+        This method scales the domain to a region (if selected) and returns weights
+        proportional to the difference between each pair of level bounds.
+
+        Parameters
+        ----------
+        domain_bounds : xr.DataArray
+            The array of bounds for the vertical domain.
+        region_bounds : Optional[np.ndarray]
+            The array of bounds for vertical selection.
+
+        Returns
+        -------
+        xr.DataArray
+            The vertical axis weights.
+        """
+        if region_bounds is not None:
+            domain_bounds = self._scale_domain_to_region(domain_bounds, region_bounds)
+
+        weights = self._calculate_weights(domain_bounds)
+        return weights
+
```

Comment:
I think this is sufficiently different from `_get_longitude_weights` (which deals with the prime meridian) and `_get_latitude_weights` (sine of bounds) to justify its own function.

Reply:
I agree